source/brailleDisplayDrivers/brailliantB.py

```diff
@@ -165,7 +165,7 @@ def __init__(self):
 			# Sometimes, a few attempts are needed to init successfully.
 			for attempt in xrange(INIT_ATTEMPTS):
 				if attempt > 0: # Not the first attempt
-					time.sleep(INIT_RETRY_DELAY) # Delay before next attempt.
+					self._dev.waitForRead(INIT_RETRY_DELAY) # Delay before next attempt.

```

Comment:
@leonardder, why did you want this to be a `waitForRead` instead of sleep? Yes, it will block the IO thread, but the IO thread shouldn't be trying to do anything else while it's initialising this display anyway. Also, there shouldn't be any data to read here; we're delaying precisely because the display isn't ready yet.

Reply:
@leonardder, why did you want this to be a waitForRead instead of sleep? Yes, it will block the IO thread, but the IO thread shouldn't be trying to do anything else while it's initialising this display anyway. 

In BDDetect, displays are initialized on the IO thread, waitFoRead will be changed as part of bdDetect to allow reading while waiting.

> Also, there shouldn't be any data to read here; we're delaying precisely because the display isn't ready yet.

Are you sure that the display simply is not ready and does not respond at all, or is it possible that it's just reacting slowly in the case of bluetooth (i.e. it takes longer than 0.2 seconds to respond and therefore passe the waitForRead at the end of the serial initialisation block?)? In the first case, waitForRead is indeed just an equivalent of sleep. In the latter case, you are wasting time with a sleep. But indeed, for HID, waitForRead doesn't make a difference.